Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -403,54 +409,113 @@ public void JumpCheck()
 				//actor.GetAnimator().SetBool("isSliding", false);
 			}
 
-			ChangeState( ActorStates.Jumping );
+			// start falling
+			StartLateJumpTimer();
+
+			ChangeState( ActorStates.Falling );
 		}
+	}
 
-		if ( !IsInState( ActorStates.Grounded ) )
+	/**
+	 * Orients the model to face the direction of movement
+	 *
+	 * Orientation is reactive to velocity, meaning the
+	 * model will face the direction of movement, rather
+	 * than the input direction.
+	 */
+	void OrientSelf()
+	{
+		Quaternion desiredLook = transform.rotation;
+		if ( IsInState( ActorStates.Climbing ) )
 		{
-			jumpColCheckTimer += Time.deltaTime;
+			Quaternion.LookRotation( climbSurfaceNormal );
 		}
+		else
+		{
+			Vector3 actualVelocity = transform.position - _lastPos;
+			actualVelocity.y = 0.0f;
+
+			Vector3 intendedVelocity = rigidbody.velocity;
+			intendedVelocity.y = 0.0f;
+
+			Vector3 lookVec = Vector3.Lerp( actualVelocity, intendedVelocity, lookDirLerp );
+
+			if ( !lookVec.IsZero() )
+			{
+				desiredLook = Quaternion.LookRotation( lookVec, transform.up );
+			}
+		}
+
+		transform.rotation = Quaternion.Lerp(
+		    transform.rotation,
+		    desiredLook,
+		    Time.deltaTime * modelTurnSpeed );
 
-		lateJumpTimer += Time.deltaTime;
-		rigidbody.velocity = Vector3.ClampMagnitude( rigidbody.velocity, maxSpeed );
+		_lastPos = transform.position;
+	}
+
+	bool IsInState( ActorStates checkState )
+	{
+		return currentState == checkState;
+	}
+
+	void SetFallSpeed( float fallSpeed )
+	{
+		Vector3 moveVec = rigidbody.velocity;
+		moveVec.y = fallSpeed;
+		rigidbody.velocity = moveVec;
 	}
 
+	#region Late Jump Timer
+	void StartLateJumpTimer()
+	{
+		_lateJump = true;
+		Invoke( "EndLateJump", lateJumpTime );
+	}
 
+	void CancelLateJumpTimer()
+	{
+		if ( IsInvoking( "EndLateJump" ) )
+		{
+			CancelInvoke( "EndLateJump" );
+		}
+		EndLateJump();
+	}
 
-	public void ModelControl()
+	void EndLateJump()
 	{
-		model.position = transform.position - modelOffset;
+		_lateJump = false;
+	}
+	#endregion
```

Comment:
Hadn't thought of using Invoke this way. Clever! We should wrap this in a Timer class and have like StartTimer, EndTimer, CheckTimer and CancelTimer with an enum as a timerID.